src/Orchard/Tasks/Locking/Services/DistributedLock.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Threading;
+
+namespace Orchard.Tasks.Locking.Services {
+
+    /// <summary>
+    /// Represents a distributed lock. />
+    /// </summary>
+    public class DistributedLock : IDisposable {
+        private IDistributedLockService _service;
+        private int _isDisposed;
+
+        private DistributedLock() {
+        }
+
+        public string Id { get; private set; }
```

Comment:
Why is this a string, shouldn't it be an int? The code on line 53 seems to be assuming so.